Yearn-agents/InstantWithdrawl/src/agent.spec.ts

```diff
@@ -0,0 +1,56 @@
+jest.useFakeTimers();
+import { HandleBlock } from "forta-agent";
+import { TestBlockEvent } from "forta-agent-tools";
+import agent from "./agent";
+import { buildWeb3, build_Mock } from "./contract.mock";
+import { generateFinding } from "./utils";
+import mockAxios from "./jest.mock";
+
+describe("Yearn: Instant Withdraw Agent", () => {
+  let handleBlock: HandleBlock;
+
+  beforeEach(async () => {
+    handleBlock = await agent.providerHandleBlock(
+      buildWeb3() as any,
+      mockAxios
+    );
+  });
+
+  describe("Finding if withdraw possible", () => {
+    it("Account withdraws: If the withdraw is working, dont return any findings", async () => {
+      const blockEvent = new TestBlockEvent();
+
+      const findings = await handleBlock(blockEvent);
+
+      expect(findings).toStrictEqual([]);
+    });
+
+    it("Another account withdraws: If the withdraw is not working, return an alert", async () => {
+      const blockEvent = new TestBlockEvent();
+
+      const findings = await handleBlock(blockEvent);
+
+      expect(findings).toStrictEqual([
+        generateFinding(
+          "2.89584189215485306814849565e+26",
+          0,
+          "0x59518884eebfb03e90a18adbaaab770d4666471e"
+        ),
+      ]);
+    });
+
+    it("Another account withdraws: If the withdraw is not working, return an alert", async () => {
+      const blockEvent = new TestBlockEvent();
+
+      const findings = await handleBlock(blockEvent);
+
+      expect(findings).toStrictEqual([
```

Comment:
And the same happens here. Is like writing a tests suite that have all the tests written in other place.

Reply:
> Is the expected behavior that the agent report lots of alerts in each block? Also in the README you mention some cases for the alerts, but the findings have none information about why are they being reported, i.e the reason of why that address is being analyzed, are the address in the top 3?, has the address 10% of the shares? <-- That kind of information.

Because this cant be fixed. The graphql API returns result according to the current stage. If I import 150 address details, they would be acc to balances across all the vaults. So some vault may have very less and won't get listed. The only way is to push the number to like a thousand to decrease the probability, but that would result in a lot of unnecessary calls and stimulation.